src/IK/include/BipedalLocomotion/IK/SE3Task.h

```diff
@@ -0,0 +1,124 @@
+
+    System::VariablesHandler::VariableDescription m_robotVelocityVariable; /**< Variable
+                                                                                  describing the
+                                                                                  robot velocity
+                                                                                  (base + joint) */
+
+    iDynTree::FrameIndex m_frameIndex; /**< Frame controlled by the OptimalControlElement */
+
+    static constexpr std::size_t m_spatialVelocitySize{6}; /**< Size of the spatial velocity vector. */
+
+    bool m_isInitialized{false}; /**< True if the task has been initialized. */
+
+public:
+
+    /**
+     * Initialize the task.
+     * @param paramHandler pointer to the parameters handler.
+     * @note the following parameters are required by the class
+     * |           Parameter Name           |   Type   |                                       Description                                      | Mandatory |
+     * |:----------------------------------:|:--------:|:--------------------------------------------------------------------------------------:|:---------:|
+     * |   `robot_velocity_variable_name`   | `string` |   Name of the variable contained in `VariablesHandler` describing the robot velocity   |    Yes    |
+     * |            `frame_name`            | `string` |                       Name of the frame controlled by the SE3Task                      |    Yes    |
+     * |             `kp_linear`            | `double` |                             Gain of the position controller                            |    Yes    |
+     * |            `kp_angular`            | `double` |                           Gain of the orientation controller                           |    Yes    |
+     * @return True in case of success, false otherwise.
+     * Where the generalized robot velocity is a vector containing the base spatial-velocity
+     * (expressed in mixed representation) and the joints velocity.
+     * @return True in case of success, false otherwise.
+     */
+    bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler> paramHandler) override;
+
+    /**
+     * Set the set of variables required by the task. The variables are stored in the
+     * System::VariablesHandler.
+     * @param variablesHandler reference to a variables handler.
+     * @note The handler must contain a variable named as the parameter
+     * `robot_velocity_variable_name` stored in the parameter handler. The variable represents the
+     * generalized velocity of the robot. Where the generalized robot velocity is a vector
+     * containing the base spatial-velocity (expressed in mixed representation) and the joints
+     * velocity.
+     * @return True in case of success, false otherwise.
+     */
+    bool setVariableHandler(const System::VariablesHandler& variablesHandler) override;
```

Comment:
I just noticed that these methods are called ``setVariableHandler`` while the object is called ``VariablesHandler`` (note the ``s``). Consider uniforming them

Reply:
Here a20a0bb6049b610ec387893b128160628b29bf5f 1e984f31ca741bc2f7c6b6ad6353b1810c5284af df7db576f70c848476d873b3182db329669cbac0 e79ad62722ea71ad14a242385140c01db78205ca b4b26e7800bf4295cdffcd8b1c024e90b1f6b891 54f28336d9eb9c5ac9233597e150f741836a4752 4c80a83e04e79fe67c381c711f6d3507054fc904 44cb5d7347bb275393c69e1f56c2d8c5a4907ba2 ccf53b79a0993be9eb3ca22a82acd228c8a371bd e3fec1aef61ab2091b6d8e0eee5dc5e235c79ea2